compiler/rustc_mir_build/src/build/expr/into.rs

```diff
@@ -255,10 +255,11 @@ impl<'a, 'tcx> Builder<'a, 'tcx> {
                         func: fun,
                         args,
                         cleanup: None,
-                        // FIXME(varkor): replace this with an uninhabitedness-based check.
-                        // This requires getting access to the current module to call
-                        // `tcx.is_ty_uninhabited_from`, which is currently tricky to do.
-                        destination: if expr.ty.is_never() {
+                        destination: if this.tcx.is_ty_uninhabited_from(
+                            this.parent_module,
```

Comment:
I think this is overly conservative - if the return type is uninhabited (even if the uninhabitedness is due to a type from another crate), then it should always be sound to omit the destination.

Reply:
Is there a better way to check uninhabitedness? I've written some code that depends on this too, so if there's a better check I'd love to use it there too.

Reply:
The rationale for taking visibility into account is to ensure that uninhabitedness remains an implementation detail unless it is visible to code in question. For example why this matters see a new test case privately-uninhabited-mir-call.rs.

Reply:
Oh, that's interesting - I didn't realize that this would affect what code we accept. In that case, taking visibility into account sounds correct. Could you add a comment explaining that?